### PR TITLE
Add Sources service and client

### PR DESCRIPTION
**Pull request by adiabramovitch**

### CODE REVIEW

**Comment by lzap** [comment]
Let's just have a call with Sources API for their advice on what exactly are we supposed to keep in our DB. I would like to put as high resistance as we can to avoid storing duplicate data, maybe we can get away just with calling their API and in-memory LUR cache.

**Comment by ezr-ondrej** [comment]
linter doesn't like extra empty lines https://pastebin.com/5V34jZPk :facepalm:
If it is not easy to get rid of those while generating, I'll find out how to not check the autogenerated files.

**Comment by adiabramovitch** [comment]
> linter doesn't like extra empty lines https://pastebin.com/5V34jZPk facepalm If it is not easy to get rid of those while generating, I'll find out how to not check the autogenerated files.

I just didn't have goimports so it didn't correct it :)

**Review by lzap — CHANGES_REQUESTED** [review]
Its allright but Makefile does not work.

I suggest to add all generated files (*.gen.go) to be skipped by all linters.

File: Makefile [hunk 1]
```diff
@@ -32,13 +32,20 @@ run:
 .PHONY: generate-clients
 generate-clients:
 	generate-ib-client
+	generate-sources-client
```
[the remark this answers is not in the document]
Comment (adiabramovitch) [reply Makefile:35]: Thank you for noticing, I will change it to the syntax you have done in your PR.

File: Makefile [hunk 2]
```diff
@@ -32,13 +32,20 @@ run:
...
 generate-ib-client:
 	cd internal/clients/image_builder
 	wget https://raw.githubusercontent.com/osbuild/image-builder/main/internal/v1/api.yaml
 	oapi-codegen -config ./configs/ib_config.yaml ./configs/ib_api.yaml
 	
+.PHONY: generate-sources-client
+generate-sources-client:
+	cd internal/clients/sources
+	wget -c https://raw.githubusercontent.com/RedHatInsights/sources-api-go/main/public/openapi-3-v3.1.json -O sources_api.json
+	oapi-codegen -config ./configs/sources_config.yml ./configs/sources_api.json
```
[the remark this answers is not in the document]
Comment (adiabramovitch) [reply Makefile:47]: so here If I want it to work for sources as well it is supposed to b like that? 

```golang
.PHONY: update-clients
update-clients:
	wget -O ./configs/ib_api.yaml -qN https://raw.githubusercontent.com/osbuild/image-builder/main/internal/v1/api.yaml
	wget -O ./configs/sources_api.json -qN https://raw.githubusercontent.com/RedHatInsights/sources-api-go/main/public/openapi-3-v3.1.json 

generate-clients: internal/clients/image_builder/client.gen.go internal/clients/sources/sources_client.gen.go

internal/clients/sources/sources_client.gen.go: configs/sources_config.yml configs/sources_api.json

internal/clients/image_builder/client.gen.go: configs/ib_config.yaml configs/ib_api.yaml
```

[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply Makefile:40]: yup, I feel like that is correct :)

[the remark this answers is not in the document]
Comment (lzap) [reply Makefile:40]: Yeah does not look great, I can improve later (with variables perhaps). Or if you like, you can introduce a variable to replace all occurrences of `internal/clients` so its a bit shorter and more readable:

```Makefile
CLIENTS_DIR := internal/clients

$(CLIENTS_DIR)/image_builder/client.gen.go: configs/ib_config.yaml configs/ib_api.yaml
# etc etc
```

**Review by ezr-ondrej — APPROVED** [review]
The config structure feels bit weird as `RestEnpoints` is hard to understand and we have double nesting for just two properties, but I guess that can be improved as we go ^_^

File: internal/config/main_config.go [hunk 3]
```diff
@@ -55,8 +55,13 @@ var config struct {
 		Environment string
 		ExitOnPanic bool
 	}
-	ImageBuilder struct {
-		URL string
+	RestEndpoints struct {
```
[the remark this answers is not in the document]
Comment (lzap) [reply internal/config/main_config.go:58]: Yeah renaming is easy in Go... I like that we at least have a common struct for these.

File: internal/config/main_config.go [hunk 4]
```diff
@@ -67,7 +72,8 @@ var Logging = &config.Logging
 var Cloudwatch = &config.Cloudwatch
 var AWS = &config.AWS
 var Features = &config.FeatureFlags
-var ImageBuilderURL = &config.ImageBuilder.URL
+var ImageBuilderURL = &config.RestEndpoints.ImageBuilder.URL
+var SourcesURL = &config.RestEndpoints.Sources.URL
```
Comment (lzap) [inline internal/config/main_config.go:76]: I would only export RestEndpoints here to follow the pattern, but it's okay.

    var RestEndpoints = &config.RestEndpoints

**Review by lzap — APPROVED** [review]
Nice.

**Pull request merged**